components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -162,33 +171,82 @@ async def do_search(
     begin_timestamp: int | None,
     end_timestamp: int | None,
     ignore_case: bool,
-    max_num_results: int,
     path_filter: str | None,
     do_count_aggregation: bool | None,
     count_by_time_bucket_size: int | None,
 ):
-    db_monitor_task = asyncio.ensure_future(
-        run_function_in_process(
-            create_and_monitor_job_in_db,
-            db_config,
-            results_cache,
-            wildcard_query,
-            tags,
-            begin_timestamp,
-            end_timestamp,
-            ignore_case,
-            max_num_results,
-            path_filter,
-            do_count_aggregation,
-            count_by_time_bucket_size,
+    if do_count_aggregation is None:
```

Comment:
Probably worth extracting this if-block into a method to keep `do_search` manageable.